v2/internal/controllers/crd_storage_storageaccount_test.go

```diff
@@ -244,3 +214,94 @@ func StorageAccount_SecretsWrittenToDifferentKubeSecrets(tc *testcommon.KubePerT
 	tc.ExpectSecretHasKeys(webSecret, "web")
 	tc.ExpectSecretHasKeys(dfsSecret, "dfs")
 }
+
+func Test_Storage_StorageAccount_ManagementPolicies(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	acct := createStorgeAccount(tc, rg)
+
+	tc.CreateResourceAndWait(acct)
+
+	tc.Expect(acct.Status.Location).To(Equal(&tc.AzureRegion))
+	expectedKind := storage.StorageAccountStatusKindStorageV2
+	tc.Expect(acct.Status.Kind).To(Equal(&expectedKind))
+
+	// Run sub-tests on storage account
+	tc.RunSubtests(
+		testcommon.Subtest{
+			Name: "Management Policies CRUD",
+			Test: func(testContext *testcommon.KubePerTestContext) {
+				StorageAccount_ManagementPolicy_CRUD(tc, acct)
+			},
+		})
+}
+
+func StorageAccount_ManagementPolicy_CRUD(tc *testcommon.KubePerTestContext, blobService client.Object) {
+	ruleEnable := true
+	daysAfterLastAccessTimeGreaterThan := 90
+	daysAfterCreationGreaterThan := 30
+	ruleType := storage.ManagementPolicyRuleTypeLifecycle
+	ruleName := "test-rule"
+
+	managementPolicy := &storage.StorageAccountsManagementPolicy{
+		ObjectMeta: tc.MakeObjectMeta("policy"),
+		Spec: storage.StorageAccountsManagementPolicies_Spec{
+			Location: tc.AzureRegion,
+			Owner:    testcommon.AsOwner(blobService),
+			Policy: &storage.ManagementPolicySchema{
+				Rules: []storage.ManagementPolicyRule{
+					{
+						Definition: &storage.ManagementPolicyDefinition{
+							Actions: &storage.ManagementPolicyAction{
+								BaseBlob: &storage.ManagementPolicyBaseBlob{
+									Delete: &storage.DateAfterModification{
+										DaysAfterLastAccessTimeGreaterThan: &daysAfterLastAccessTimeGreaterThan,
+									},
+								},
+								Version: &storage.ManagementPolicyVersion{
+									Delete: &storage.DateAfterCreation{
+										DaysAfterCreationGreaterThan: &daysAfterCreationGreaterThan,
+									},
+								},
+							},
+							Filters: &storage.ManagementPolicyFilter{
+								BlobTypes:   []string{"blockBlob"},
+								PrefixMatch: []string{"sample-container/blob1"},
+							},
+						},
+						Enabled: &ruleEnable,
+						Name:    &ruleName,
+						Type:    &ruleType,
+					},
+				},
+			},
+		},
+	}
+
+	tc.CreateResourceAndWait(managementPolicy)
+	defer tc.DeleteResourceAndWait(managementPolicy)
+}
+
+func createStorgeAccount(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup) *storage.StorageAccount {
```

Comment:
maybe call this `newStorageAccount`, as it's making a `*storage.StorageAccount` but not actually creating it (no k8s client call)